modules/ding_mobilesearch/ding_mobilesearch.module

```diff
@@ -0,0 +1,504 @@
+<?php
+
+/**
+ * @file
+ * Mobilesearch logic.
+ */
+
+include_once 'includes/trigger.inc';
+include_once 'lib/bootstrap.inc';
+include_once 'plugins/rest.inc';
+
+/**
+ * Implements hook_permission().
+ */
+function ding_mobilesearch_permission() {
+  return array(
+    'administer mobilesearch' => array(
+      'title' => t('Administer mobilesearch'),
+    ),
+  );
+}
+
+/**
+ * Implements hook_menu().
+ */
+function ding_mobilesearch_menu() {
+  $items = array();
+
+  $items['admin/config/ding/mobilesearch/content-export'] = array(
+    'title' => 'Content export',
+    'description' => 'Content exporting settings.',
+    'page callback' => 'drupal_get_form',
+    'page arguments' => array('ding_mobilesearch_settings'),
+    'access arguments' => array('administer mobilesearch'),
+    'file' => 'includes/admin.inc',
+  );
+
+  $items['admin/config/ding/mobilesearch/content-export/settings'] = array(
+    'title' => 'Settings',
+    'type' => MENU_DEFAULT_LOCAL_TASK,
+    'weight' => -10,
+  );
+
+  $items['admin/config/ding/mobilesearch/content-export/plugin'] = array(
+    'title' => 'Plugins',
+    'description' => 'Set a special configuration options for plugins.',
+    'page callback' => 'drupal_get_form',
+    'page arguments' => array('ding_mobilesearch_plugin_settings'),
+    'access arguments' => array('administer mobilesearch'),
+    'type' => MENU_LOCAL_TASK,
+    'weight' => 1,
+    'file' => 'includes/admin.inc',
+  );
+
+  return $items;
+}
+
+/**
+ * Implements hook_theme().
+ */
+function ding_mobilesearch_theme($existing, $type, $theme, $path) {
+  return array(
+    'ding_mobilesearch_types_table' => array(
+      'render element' => 'form',
+      'file' => 'includes/admin.inc',
+    ),
+    'ding_mobilesearch_menus_table' => array(
+      'render element' => 'form',
+      'file' => 'includes/admin.inc',
+    ),
+  );
+}
+
+/**
+ * Implements hook_mobilesearch_plugin_info().
+ */
+function ding_mobilesearch_mobilesearch_plugin_info() {
+  return array(
+    'mobilesearch_rest' => array(
+      'title' => t('REST plugin'),
+      'description' => t('Export data to Mongo through REST API.'),
+      'exec_callback' => 'mobilesearch_rest_exec',
+      'config_form_callback' => 'mobilesearch_rest_config_form',
+      'config_form_validate' => 'mobilesearch_rest_config_form_validate',
+      'node_form_callback' => 'mobilesearch_rest_node_form',
+    ),
+  );
+}
+
+/**
+ * Implements hook_form_FORM_ID_alter().
+ */
```

Comment:
It might be my, but I would really like some comments about the alter to menu editing. It loads a vocabulary (which i'am not sure exists in DDB CMS, might be defined in this PR later on).

So you can define menu items based on agency id ?